docs/reference/export-logs.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: Exporting Logs (OpenTelemetry)
+sidebar_position: 1200
+description: Using OpenTelemetry to export gadget logs
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Inspektor Gadget supports [exporting logs to OpenTelemetry](https://opentelemetry.io/docs/specs/otel/logs/) using the
+otlp-grpc exporter. You can log the events of any datasource by configuring exporters in the `operator.otel-logs`
+section of the config file like so:
+
+```yaml
+operator:
+  otel-logs:
+    exporters:
+      my-log-exporter:
+        exporter: otlp-grpc
+        compression: gzip
+        endpoint: "127.0.0.1:4317"
+        insecure: true
+```
+
+This will configure an exporter named `my-log-exporter` with the given endpoint, gzip compression enabled and TLS
+disabled.
+
+You can then run a gadget and activate the exporter for it by setting the `--otel-logs-exporter=my-log-exporter` flag.
+
+### Exporter settings
+
+#### exporter
+
+Currently we only support `otlp-grpc`.
+
+#### compression
+
+Compression can be set to either "none" (no compression) or "gzip" (gzip compression).
+
+#### endpoint
+
+IP address and port of the gRPC receiver.
+
+#### insecure
+
+If set to true, the gRPC connection will not use TLS encryption. False by default.
+
+## Annotations
+
+Annotations can be used to define how logs should be generated from a datasource. Let's look at an example:
+
+```yaml
+datasources:
```

Comment:
(trying to understand the flow) Lets say I have use-case in which I want to generate logs for failing DNS requests. I will:
- Annotate the gadget.yaml with the log format/severity I am interested in. (do I need to maintain custom version of gadget?)
- Run the gadget with a specific filter `--filte RCODE!=Success` .

Or I should do everything in instance manifest file? Just trying to understand which of these would be recommended!

Reply:
If you define a filter like `--filter RCODE!=Success`, then you probably want to set the severity of the logs globally to an error value, for example. If you just log every DNS request, then you can't differentiate between severities unless you add a specific field for that in the eBPF part (so that it can be extracted and be dynamic instead of static.)

Instead of annotating in the gadget.yaml, you can also use the `--annotate` flag and define the annotations ad-hoc; this way you don't have to create your own version of a gadget.

Reply:
> If you define a filter like --filter RCODE!=Success, then you probably want to set the severity of the logs globally to an error value, for example. If you just log every DNS request, then you can't differentiate between severities

I see, I think is fine to keep it that way! 

> unless you add a specific field for that in the eBPF part (so that it can be extracted and be dynamic instead of static.)

Just trying to understand the idea. You mean being able to set `logs.severity` based on a certain field. We don't have anything that can glue this together right now. Right?

> Instead of annotating in the gadget.yaml, you can also use the --annotate flag and define the annotations ad-hoc; this way you don't have to create your own version of a gadget.

Sounds good!

Reply:
> Just trying to understand the idea. You mean being able to set logs.severity based on a certain field. We don't have anything that can glue this together right now. Right?

Yeah - currently no gadget is doing that. As "glue", maybe it could make sense in the future to also allow an expr-lang expression that returns a numeric result for the severity value.